base_tier_validation/models/tier_validation.py

```diff
@@ -184,37 +154,61 @@ def _get_validated_message(self):
+        return self.validation_status == "validated" and msg or ""
 
     def _get_rejected_message(self):
         msg = """<i class="fa fa-thumbs-down" /> %s""" % _(
             """Operation has been <b>rejected</b>."""
         )
-        return self.rejected and msg or ""
+        return self.validation_status == "rejected" and msg or ""
+
+    @api.depends("validation_status")
+    def _compute_to_validate_message(self):
+        for rec in self:
+            rec.to_validate_message = rec._get_to_validate_message()
+
+    def _validated_states(self):
+        """Override for different validation policy."""
+        return ["approved"]
 
-    def _compute_validated_rejected(self):
+    @api.depends("validation_status")
+    def _compute_validated_message(self):
         for rec in self:
-            rec.validated = self._calc_reviews_validated(rec.review_ids)
             rec.validated_message = rec._get_validated_message()
-            rec.rejected = self._calc_reviews_rejected(rec.review_ids)
+
+    def _rejected_states(self):
+        """Override for different rejected policy."""
+        return ["rejected"]
+
+    @api.depends("validation_status")
+    def _compute_rejected_message(self):
+        for rec in self:
             rec.rejected_message = rec._get_rejected_message()
-            rec.to_validate_message = rec._get_to_validate_message()
 
+    @api.depends("review_ids", "review_ids.status")
     def _compute_validation_status(self):
+        validated_states = self._validated_states()
+        rejected_states = self._rejected_states()
         for item in self:
-            if item.validated and not item.rejected:
+            reviews = item.review_ids
+            any_approved = any(reviews.filtered(lambda x: x.status in validated_states))
+            any_rejected = any(reviews.filtered(lambda x: x.status in rejected_states))
+            any_pending = any(reviews.filtered(lambda x: x.status == "pending"))
+            if (
+                reviews
+                and not any(
+                    reviews.filtered(lambda x: x.status not in validated_states)
+                )
+                and not any_rejected
+            ):
                 item.validation_status = "validated"
-            elif not item.validated and item.rejected:
+            elif reviews and any_rejected:
                 item.validation_status = "rejected"
-            elif (
-                not item.validated
-                and not item.rejected
-                and any(item.review_ids.filtered(lambda x: x.status == "pending"))
-            ):
+            elif reviews and not any_approved and not any_rejected and any_pending:
                 item.validation_status = "pending"
             elif (
-                not item.validated
-                and not item.rejected
+                not any_approved
+                and not any_rejected
                 and any(item.review_ids.filtered(lambda x: x.status == "waiting"))
```

Comment:
Can also be simplified. Maybe also add an any_waiting variable.
```suggestion
                any(item.review_ids.filtered(lambda x: x.status == "waiting"))
```